### PR TITLE
Feature/improve lib

### LINKED ISSUE #33: Custom marks/blocks can't be cast when using ExProsemirror Ecto helpers

### PR DESCRIPTION
**Pull request by Matsa59**
## 📖 Description

Close #33 #26

## 📋 Type of change

- [x] New feature (with breaking change)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

Doc should be rewrite also ....

### CODE REVIEW

File: lib/ex_prosemirror/config.ex [hunk 1]
```diff
@@ -101,23 +47,12 @@ defmodule ExProsemirror.Config do
...
+      [inline: true, marks: [:strong], blocks: [{:heading, [:h1, :h2]}, :p]]
 
   """
-  def load(type, custom_config \\ [])
-
-  def load(:default, custom_config) do
-    put_default_types()
-    |> override(custom_config)
-  end
+  def load(type)
```
Comment (tomkonidas) [inline lib/ex_prosemirror/config.ex:53]:
```suggestion
```
Do we still need this? we only have one `load/1` defined

File: lib/ex_prosemirror/type.ex [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  """
+
+  import Ecto.Changeset
+  import ExProsemirror.SchemaHelper
+
+  require ExProsemirror.TypeGenerator
+  ExProsemirror.TypeGenerator.generate_all()
+
+  @doc """
+  Changeset use by all `ExProsemirror.Type`.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/type.ex:13]:
```suggestion
  Changeset used by all `ExProsemirror.Type`.
```

File: test/form_test.exs [hunk 3]
```diff
@@ -39,68 +39,51 @@ defmodule ExProsemirror.FormTest do
...
       assert input_html =~
-               ~s(data-blocks="[&quot;h1&quot;,&quot;p&quot;]")
+               ~s(<div class="ex-prosemirror" data-blocks="[]" data-inline="false" data-marks="[]" data-target="#title" id="title")
     end
-  end
 
-  describe "test prosemirror_input" do
-    test "without options" do
-      form = %Form{data: %{title_plain: "hello"}}
+    test "unkow type should log a warn" do
```
Comment (tomkonidas) [inline test/form_test.exs:75]:
```suggestion
    test "unknown type should log a warn" do
```

File: test/form_test.exs [hunk 4]
```diff
@@ -39,68 +39,51 @@ defmodule ExProsemirror.FormTest do
...
+      log =
+        capture_log(fn ->
+          %Form{data: %{title_plain: "hello"}}
+          |> prosemirror_editor(:title, type: :unkown)
+          |> safe_to_string()
+        end)
 
-      assert inputs_html =~ hidden_input_html
-      assert inputs_html =~ editor_input_html
+      assert log =~ ~S(ExProsemirror - Type "unkown" not found)
```
Comment (tomkonidas) [inline test/form_test.exs:85]:
```suggestion
      assert log =~ ~S(ExProsemirror - Type "unknown" not found)
```

**Review by tomkonidas — APPROVED** [review]

**Review by onixus74 — APPROVED** [review]
Awesome stuff 👍

**Comment by Matsa59** [comment]
Also, important to consider that this will introduce some technical debt about docs and the system to transform prosemirror structs to json.

I'll open 2 issues to track those debts

**Pull request merged**